lib/transaction/transaction.js

```diff
@@ -838,6 +842,7 @@ Transaction.prototype._getInputAmount = function() {
 };
 
 Transaction.prototype._updateChangeOutput = function() {
+  this._hashCache = false;
```

Comment:
Shouldn't we set hashCache  to false in other function, like: .sign?

Reply:
let's take the hash cache part out of this PR - need to discuss plan on that.

Reply:
Good idea!